tests/rkt_api_service_test.go

```diff
@@ -0,0 +1,297 @@
+	// the lengths of the two lists are equal.
+	for _, m := range rawNets {
+		found := false
+		for _, n := range apiNets {
+			if m.name == n.Name {
+				found = true
+				if m.ipv4 != n.Ipv4 {
+					t.Errorf("Expected %q, saw %q", m.ipv4, n.Ipv4)
+				}
+				break
+			}
+		}
+		if !found {
+			t.Errorf("Expected network (name: %q, ipv4: %q) in networks", m.name, m.ipv4)
+		}
+	}
+}
+
+// Check the pod's information by 'rkt status'.
+func checkPod(t *testing.T, ctx *rktRunCtx, p *v1alpha.Pod, checkAppState bool) {
+	podInfo := getPodInfo(t, ctx, p.Id)
+	if podInfo.id != p.Id {
+		t.Errorf("Expected %q, saw %q", podInfo.id, p.Id)
+	}
+	if podInfo.pid != int(p.Pid) {
+		t.Errorf("Expected %q, saw %q", podInfo.pid, p.Pid)
+	}
+	checkPodState(t, podInfo.state, p.State)
+	checkPodApps(t, podInfo.apps, p.Apps, checkAppState)
+	checkPodNetworks(t, podInfo.networks, p.Networks)
+
+	// TODO(yifan): Check manifest only when testing InspectPod().
+	if p.Manifest != nil && !bytes.Equal(podInfo.manifest, p.Manifest) {
+		t.Errorf("Expected %q, saw %q", string(podInfo.manifest), string(p.Manifest))
+	}
+}
+
+// Check the image's information by 'rkt image list'.
+func checkImage(t *testing.T, ctx *rktRunCtx, m *v1alpha.Image) {
+	imgInfo := getImageInfo(t, ctx, m.Id)
+	if imgInfo.id != m.Id {
+		t.Errorf("Expected %q, saw %q", imgInfo.id, m.Id)
+	}
+	if imgInfo.name != m.Name {
+		t.Errorf("Expected %q, saw %q", imgInfo.name, m.Name)
+	}
+	if imgInfo.version != m.Version {
+		t.Errorf("Expected %q, saw %q", imgInfo.version, m.Version)
+	}
+	if imgInfo.importTime != m.ImportTimestamp {
+		t.Errorf("Expected %q, saw %q", imgInfo.importTime, m.ImportTimestamp)
+	}
+
+	// TODO(yifan): Check manifest only when testing InspectImage().
+	if m.Manifest != nil && !bytes.Equal(imgInfo.manifest, m.Manifest) {
+		t.Errorf("Expected %q, saw %q", string(imgInfo.manifest), string(m.Manifest))
+	}
+}
+
+func TestAPIServiceGetInfo(t *testing.T) {
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+
+	svc := startAPIService(t, ctx)
+	defer stopAPIService(t, svc)
+
+	conn, err := grpc.Dial("localhost:15441", grpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	defer conn.Close()
+	c := v1alpha.NewPublicAPIClient(conn)
+
+	resp, err := c.GetInfo(context.Background(), &v1alpha.GetInfoRequest{})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	expectedAPIVersion := "1.0.0-alpha"
+	if resp.Info.ApiVersion != expectedAPIVersion {
+		t.Errorf("Expected api version to be %q, but saw %q", expectedAPIVersion, resp.Info.ApiVersion)
+	}
+}
+
+func TestAPIServiceListInspectPods(t *testing.T) {
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+
+	svc := startAPIService(t, ctx)
+	defer stopAPIService(t, svc)
+
+	conn, err := grpc.Dial("localhost:15441", grpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	defer conn.Close()
+	c := v1alpha.NewPublicAPIClient(conn)
```

Comment:
The lines above are repeated for each test. Maybe put the initialization into some struct or function?